src/main/java/uk/gov/ons/ctp/response/action/service/decorator/PartyAndContact.java

```diff
@@ -15,88 +15,101 @@
     actionContact.setRuName(businessUnitAttributes.getName());
     actionContact.setTradingStyle(generateTradingStyle(businessUnitAttributes));
-
-    if (context.getSampleUnitType().equals(SampleUnitDTO.SampleUnitType.BI)) {
-      decorateBIcase(actionRequest, context, actionContact);
-    } else if (isNotifyType(context)) {
-      decorateBCaseNotifyType(respondentParties, actionRequest, actionContact);
+    if (isNotifyType(context)) {
+      decorateNotifyType(respondentParties, actionRequest, actionContact);
     } else if (isPrinterType(context)) {
-      decorateBCasePrinterType(context, actionRequest, actionContact);
+      decoratePrinterType(respondentParties, actionRequest, actionContact);
     } else {
       throw new IllegalStateException(INVALID_SAMPLE_UNIT_TYPE_AND_HANDLER);
     }
-
     actionRequest.setContact(actionContact);
-    actionRequest.setEnrolmentStatus(getEnrolmentStatus(businessParty));
   }
 
-  private void decorateBIcase(
-      ActionRequest actionRequest, ActionRequestContext context, ActionContact contact) {
-    PartyDTO respondentParty = context.getChildParties().get(0);
-    actionRequest.setRespondentStatus(respondentParty.getStatus());
-    Attributes respondentAttributes = respondentParty.getAttributes();
-    populateContactDetails(respondentAttributes, contact);
-  }
+  private String getEnrolmentStatus(final PartyDTO parentParty) {
+    final List<String> enrolmentStatuses = new ArrayList<>();
 
-  private void decorateBCaseNotifyType(
-      List<PartyDTO> respondentParties, ActionRequest actionRequest, ActionContact actionContact) {
-    PartyDTO respondentParty = respondentParties.get(0);
-    actionRequest.setRespondentStatus(respondentParty.getStatus());
+    final List<Association> associations = parentParty.getAssociations();
+    if (associations != null) {
+      for (Association association : associations) {
+        for (Enrolment enrolment : association.getEnrolments()) {
+          enrolmentStatuses.add(enrolment.getEnrolmentStatus());
+        }
+      }
+    }
 
-    final Attributes respondentAttributes = respondentParty.getAttributes();
-    populateContactDetails(respondentAttributes, actionContact);
-  }
+    String enrolmentStatus = null;
 
-  private void decorateBCasePrinterType(
-      ActionRequestContext context, ActionRequest actionRequest, ActionContact actionContact) {
-    actionRequest.setRespondentStatus(parseRespondentStatuses(context.getChildParties()));
-    List<PartyDTO> createdChildParties =
-        filterListByStatus(context.getChildParties(), ActionProcessingService.CREATED);
+    if (enrolmentStatuses.contains(ActionProcessingService.PENDING)) {
+      enrolmentStatus = ActionProcessingService.PENDING;
+    }
 
-    if (createdChildParties != null && createdChildParties.size() > 0) {
-      actionRequest.setIac(""); // Don't want to send this to the business, breaks if null
+    if (enrolmentStatuses.contains(ActionProcessingService.ENABLED)) {
```

Comment:
it would be more efficient to do this test first, then put the second test in an `else if` so that it doesn't waste time doing two string comparisons and two variable assignments, when it could only do one of each.